src/Core/Addon/Theme/ThemeChecker.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * 2007-2015 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2015 PrestaShop SA
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShop\PrestaShop\Core\Addon\Theme;
+
+class ThemeChecker
+{
+    private $theme;
+
+    public function setTheme(Theme $theme)
```

Comment:
Why not pass the `$theme` directly to `isValid()`?

I like to avoid putting state in objects as much as possible. Most evil in programming comes from state.

Also, imagine tomorrow your `ThemeChecker` starts depending on many other services (maybe it will need to communicate with the Addons API for instance).

If it holds a reference to the `$theme` then you can't put this `ThemeChecker` service inside a service container anymore and you will need to `new` it all over the place.

Whereas if `$theme` is passed to `isValid` you can share a single instance of `ThemeChecker` to validate all of your themes.